### PR TITLE
Remove webdav and other unstandardized methods

### PR DESCRIPTION
**Pull request by guoye-zhang**
Keeping only standard methods and ones we use https://developer.mozilla.org/en-US/docs/Web/HTTP/Methods

### CODE REVIEW

File: Sources/HTTPTypes/HTTPRequest.swift [hunk 1]
```diff
@@ -227,72 +227,22 @@ extension HTTPRequest: CustomDebugStringConvertible {
...
-    /// UNSUBSCRIBE
-    public static var unsubscribe: Self { .init(unchecked: "UNSUBSCRIBE") }
-    /// SOURCE
-    public static var source: Self { .init(unchecked: "SOURCE") }
+    /// TRACE
+    public static var trace: Self { .init(unchecked: "TRACE") }
+    /// PATCH
+    public static var patch: Self { .init(unchecked: "PATCH") }
     /// CONNECT-UDP
     static var connectUDP: Self { .init(unchecked: "CONNECT-UDP") }
```
Comment (FranzBusch) [inline Sources/HTTPTypes/HTTPRequest.swift:247]: What's with this one? I don't see it mentioned in the link you provided.

Comment (guoye-zhang) [reply Sources/HTTPTypes/HTTPRequest.swift:247]: This isn't a public method. It's used internally for talking to Private Relay servers implementing an older draft:

https://datatracker.ietf.org/doc/html/draft-ietf-masque-connect-udp-04

Once every server moves to the RFC version (CONNECT), this constant can be removed.

Comment (ekinnear) [reply Sources/HTTPTypes/HTTPRequest.swift:247]: CONNECT-UDP was used publicly in early drafts ([-04](https://datatracker.ietf.org/doc/html/draft-ietf-masque-connect-udp-04) and earlier) for the CONNECT-UDP work in MASQUE which is now [RFC9298](https://www.rfc-editor.org/rfc/rfc9298) and uses Extended CONNECT instead. Once usage of those draft versions tails off, we should remove it.

**Comment by ekinnear** [comment]
Just to confirm, removing these doesn't mean that anyone who needs to use one can't use it -- it should be easy enough to bring whichever ones are needed, so this is just cleaning up the built-in list.

**Review by ekinnear — APPROVED** [review]

**Comment by guoye-zhang** [comment]
> Just to confirm, removing these doesn't mean that anyone who needs to use one can't use it -- it should be easy enough to bring whichever ones are needed, so this is just cleaning up the built-in list.

Yeah, you can always create a custom method `HTTPRequest.Method("METHOD")`

**Pull request merged**